packages/core/src/timeline.ts

```diff
@@ -103,6 +113,21 @@ export class Timeline {
             } else {
               await pluginResult;
             }
+
+            if ([PluginType.before, PluginType.enrichment].includes(type)) {
+              Inspector.trace({
+                timestamp: new Date().toISOString(),
+                stage: 'enriched',
```

Comment:
How about triggering this for all plugin types, switch the `stage` depending on the type.

As for the name of the plugin it could pull the `key` for the Destinations ([it's the only one that is enforced at the moment](https://github.com/segmentio/analytics-react-native/blob/e1dfeb8ad852350a902e4255bf7525a8af205fe6/packages/core/src/plugin.ts#L117)):

```ts
let stageMap = {
  PluginType.enrichment: "enriched",
  PluginType.before: "enriched",
  PluginType.destination: "delivered",
  PluginType.after: "after",
// ...
};
const stage = stageMap[type];

 Inspector.trace({
	timestamp: new Date().toISOString(),
	stage: stage,
    event: result as never,
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    id: event.messageId!,
    enricher: {
	 name: (type === PluginType.destination) ? plugin.key : undefined,
      type: PluginType[type],
    },
})
```